torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -1212,20 +1459,71 @@ Error PyTorchModelLoader::loadQuantizedAddRelu(const torch::jit::Node *ptNode) {
 Error PyTorchModelLoader::loadQuantizedLinear(const torch::jit::Node *ptNode) {
   auto inputs = ptNode->inputs();
   auto outputs = ptNode->outputs();
-  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, 5, outputs, 1));
+  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, 4, outputs, 1));
 
   glow::NodeValue input;
   ASSIGN_VALUE_OR_RETURN_ERR(
       input, getGlowNodeValueForValue(inputs[QuantizedLinearInputs::input]));
   input = rescaleUIntToInt(input);
 
-  glow::NodeValue weight;
+  glow::NodeValue packedWeight;
   ASSIGN_VALUE_OR_RETURN_ERR(
-      weight, getGlowNodeValueForValue(inputs[QuantizedLinearInputs::weight]));
+      packedWeight,
+      getGlowNodeValueForValue(inputs[QuantizedLinearInputs::packed_weights]));
+
+  glow::Constant *packedWeightConstant =
+      llvm::dyn_cast<glow::Constant>(packedWeight.getNode());
+  RETURN_ERR_IF_NOT(packedWeightConstant,
+                    "quantized::linear packed weights must be constant");
+  const glow::Tensor &packedWeightTensor =
+      packedWeightConstant->getPayload().getHandle<bool>().clone();
+
+  auto ptTensor = glowTensorToPTTensor(packedWeightTensor, at::kByte);
+
+  auto unpackedParams = glow::fbgemm_linear_unpack(ptTensor);
+  const at::Tensor &ptWeightTensor = std::get<0>(unpackedParams);
+  const c10::optional<at::Tensor> &ptBiasTensor = std::get<1>(unpackedParams);
+
+  bool isRowwiseQuantized = ptWeightTensor.is_quantized() &&
```

Comment:
same thing, are both checks necessary or should we assert that one implies the other?

Reply:
In QuantizedLinear, we dont have group so we will only use this to indenty a channel wise quantized tensor.
Also, both of them are needed.